### PR TITLE
feat(install): add functionality to install CSPC operator

### PR DESCRIPTION
**Pull request by sagarkrsd**
This PR adds the functionality to install cStor operator related CRDs, deployments, service account, cluster role and cluster role binding which will be used for cStor pool provisioning.
 
The new fields that has been introduced the OpenEBS CR yaml for configuring cStor-operators installation are displayed inside cstorConfig:
```yaml
# This configuration is for OpenEBS Installation, ControlPlaneUpgrade
# and unInstallation.
apiVersion: dao.mayadata.io/v1alpha1
kind: OpenEBS
metadata:
  name: install-openebs-1.9.0
  # Namespace i.e. the namespace where the openebs-upgrade operator and the
  # other openebs components will be installed/needs-to-be-installed.
  namespace: openebs
  labels:
    name: openebs-upgrade
spec:
  # OpenEBS Version to be installed.
  version: "1.9.0"

  # cstorConfig stores the configuration for Cstor: CAS Data Engine
  #
  # The primary function of cStor is to serve the iSCSI block storage using
  # the underlying disks in a cloud native way.cStor is a very light weight
  # and feature rich storage engine. It provides enterprise grade features
  # such as synchronous data replication, snapshots, clones, thin provisioning
  # of data, high resiliency of data, etc.
  #
  # It has two main components: cStor pool pods and cStor target pods.
  #
  # pool, poolMgmt, target and volumeMgmt are the containers which are deployed
  # in the k8s cluster.
  cstorConfig:
    pool:
      imageTag:
    poolMgmt:
      imageTag:
    target:
      imageTag:
    volumeMgmt:
      imageTag:
    # NOTE: This PR introduces the below new fields for configuring cStor-operators installation.
    #
    # Below fields have been introduced from 1.9.0 version onwards in order to support
    # installation of cStor data engine operators and the related components which will help
    # in cStor pool provisioning.
    # CstorPoolCluster(CSPC), CstorPoolInstance(CSPI), CSPC-operator are the new schema
    # introduced for cStor pool provisioning.
    #
    # Note: All the fields introduced below are completely optional i.e., if user
    # doesn't add any of them, all the CSPC related components will be installed with
    # default values.
    cspiMgmt:
      # example image for cspiMgmt -> "quay.io/openebs/cstor-pool-manager:1.9.0"
      # this field should only contain the image tag, for example, 1.9.0
      imageTag:
    volumeManager:
      # example image for volumeManager -> "quay.io/openebs/cstor-volume-manager:1.9.0"
      # this field should only contain the image tag, for example, 1.9.0
      imageTag:
    cspcOperator:
      enabled:
      # example image for cspc-operator -> "quay.io/openebs/cspc-operator:1.9.0"
      # this field should only contain the image tag, for example, 1.9.0
      imageTag:
      replicas:
      resources:
      nodeSelector:
      tolerations:
      affinity:
    cvcOperator:
      enabled:
      # example image for cspc-operator -> "quay.io/openebs/cvc-operator:1.9.0"
      # this field should only contain the image tag, for example, 1.9.0
      imageTag:
      replicas:
      resources:
      nodeSelector:
      tolerations:
      affinity:
```

Note: For taking a detailed look at the OpenEBS CR YAML, please take a look at `deploy/example.yaml` file.

Signed-off-by: sagarkrsd <sagar.kumar@mayadata.io>

### CODE REVIEW

File: controller/openebs/cstor.go [hunk 1]
```diff
@@ -50,5 +61,199 @@ func (p *Planner) setCStorDefaultsIfNotSet() error {
...
+				p.ObservedOpenEBS.Spec.Policies.Monitoring.Image, "spec", "value")
+		}
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+	updateContainer := func(obj *unstructured.Unstructured) error {
+		containerName, _, err := unstructured.NestedString(obj.Object, "spec", "name")
```
Comment (shovanmaity) [inline controller/openebs/cstor.go:225]: n_: can we use same package
unstructured -> unstruct

File: controller/openebs/cstor.go [hunk 2]
```diff
@@ -50,5 +61,199 @@ func (p *Planner) setCStorDefaultsIfNotSet() error {
...
+				return err
+			}
+		}
+		err = unstructured.SetNestedSlice(obj.Object, envs, "spec", "env")
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	err = unstruct.SliceIterator(containers).ForEachUpdate(updateContainer)
```
Comment (shovanmaity) [inline controller/openebs/cstor.go:176]: n_: can we use same package
unstructured -> unstruct

File: controller/openebs/cstor.go [hunk 3]
```diff
@@ -50,5 +61,199 @@ func (p *Planner) setCStorDefaultsIfNotSet() error {
...
+				p.ObservedOpenEBS.Spec.Policies.Monitoring.Image, "spec", "value")
+		}
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+	updateContainer := func(obj *unstructured.Unstructured) error {
+		containerName, _, err := unstructured.NestedString(obj.Object, "spec", "name")
```
Comment (shovanmaity) [inline controller/openebs/cstor.go:153]: n_: can we use same package
unstructured -> unstruct

**Review by shovanmaity — APPROVED** [review]

File: controller/openebs/cstor.go (hunk 2 above)
Comment (sagarkrsd) [reply controller/openebs/cstor.go:176]: This is unstruct only, right?

**Review by AmitKumarDas — APPROVED** [review]

File: controller/openebs/cstor.go (hunk 2 above)
Comment (shovanmaity) [reply controller/openebs/cstor.go:176]: sorry wrong place

**Pull request merged**